bin/asl-say

```diff
@@ -43,60 +45,72 @@ function build_and_say(){
 }
 
 function get_time(){
-	
+
 	# calculate the hour
-	NOW_H=$(date +%I | sed -e 's/0//g')
-
-	# calculate the minute audio	
-	NOW_M=$(date +%M | sed -e s/'^0//')
-	NOW_FM=$(( ${NOW_M} / 10 ))
-	case ${NOW_FM} in
-		0)	NOW_FM=0 ;;
-		1)	NOW_FM=$(date +%M) ;;
-		*)	NOW_FM="${NOW_FM}0" ;;
-	esac
-	NOW_SM=$(( ${NOW_M} % 10 ))
-	if [ $NOW_FM -eq 0 ] && [ $NOW_SM -eq 0 ]; then
-		$NOW_FM=oclock
-		$NOW_SM=""
+	HOUR=$($DATE_CMD +%I | sed -e 's/0//g')
+	HOUR_FULL="digits/${HOUR}"
+
+	# calculate the minute audio
+	MINUTE=$($DATE_CMD +%0M | sed -e s/'^0//')
+	if (( ${MINUTE} == 0 )); then
+		MINUTE_FULL="digits/oclock"
+	elif (( ${MINUTE} < 10 )); then
+		MINUTE_FULL="digits/oh digits/${MINUTE}"
+	elif (( ${MINUTE} < 20 )); then
+		MINUTE_FULL="digits/${MINUTE}"
+	else
+		A=$(( ${MINUTE} / 10 ))
+		B=$(( ${MINUTE} % 10 ))
+		if (( $B == 0 )); then
+			MINUTE_FULL="digits/${A}0"
+		else
+			MINUTE_FULL="digits/${A}0 digits/${B}"
+		fi
 	fi
-	[ $NOW_FM -ge 1 ] && [ $NOW_FM -lt 20 ] && NOW_SM=""
 
 	# am/pm
-	NOW_A=$(date +%P | sed 's/^\([ap]\)\(m\)/\1-\2/')
+	AM_PM=$($DATE_CMD +%P | sed 's/^\([ap]\)\(m\)/\1-\2/')
 
 	# timezone with letters prefix
-	NOW_TZ=$(date +%Z | tr '[A-Z]' '[a-z]' | sed 's/./letters\/& /g')
-	echo "rpt/thetimeis digits/${NOW_H} digits/${NOW_FM} \ 
-		digits/${NOW_SM} digits/${NOW_A} ${NOW_TZ}"
+	TIMEZONE=$($DATE_CMD +%Z | tr '[A-Z]' '[a-z]' | sed 's/./letters\/& /g')
+
+	echo "rpt/thetimeis ${HOUR_FULL} ${MINUTE_FULL} digits/${AM_PM} ${TIMEZONE}"
 }
 
 function get_time24(){
 
-	NOW_HR=$(date +%-H)
-	if (( 0 <= ${NOW_HR} && ${NOW_HR} < 19 )); then
-		NOW_HR_FULL="digits/${NOW_HR} hours"		
-	elif (( 20 <= ${NOW_HR } && ${NOW_HR} < 24 )); then
-		A=$(( ${NOW_HR} / 10 ))
-		B=$(( ${NOW_HR} % 10 ))
-		NOW_HR_FULL="digits/${A} digits/${B} hours" 
+	# calculate the hour audio
+	HOUR=$($DATE_CMD +%-H)
+	if (( ${HOUR} < 20 )); then
+		HOUR_FULL="digits/${HOUR} hours"
```

Comment:
What are people's expectations on 24 hour time? Right now it is saying "21 hours 9 minutes". 
I am used to hearing the 24 hour clock as "21 hundred oh 9"

Reply:
... and I just reworked the "time24" handling.